donkeycar/parts/telemetry.py

```diff
@@ -85,18 +103,41 @@ def publish(self):
 
         if not packet:
             return
-
+            
         if self._use_json_format:
             packet = [{'ts': k, 'values': v} for k, v in packet.items()]
             payload = json.dumps(packet)
 
-            self._mqtt_client.publish(self._topic, payload)
+            try:
+                self._mqtt_client.publish(self._topic, payload)
+            except:
+                logger.error('Error publishing topic "{}"'.format(topic))
             # print(f'Total updates - {self._total_updates} (payload size={len(payload)})')
         else:
-            # Publish only the last timestamp
+            # Publish only the last timestamp for per step metrics
             last_sample = packet[list(packet)[-1]]
             for k, v in last_sample.items():
-                self._mqtt_client.publish('{}/{}'.format(self._topic, k), v)
+                if k in self._step_inputs:
+                    topic = '{}/{}'.format(self._topic, k)
+                    
+                    try:
+                        # Remove unsupported numpy types
+                        if isinstance(v, np.generic):
+                            v =  np.asscalar(v)
+
+                        self._mqtt_client.publish('{}/{}'.format(self._topic, k), v)
+                    except TypeError:
+                        logger.error('Cannot publish topic "{}" with value of type {}'.format(topic, type(v)))
+                    except:
+                        logger.error('Error publishing topic "{}"'.format(topic))
+                    
+            # Publish all logs
+            for tm, sample in packet.items():
+                if LOG_MQTT_KEY in sample:
+                    try:
+                        self._mqtt_client.publish('{}/{}'.format(self._topic, LOG_MQTT_KEY), sample[LOG_MQTT_KEY])
+                    except:
```

Comment:
Better to print the exception here, so you know what went wrong:

```python
except Exception as e:
    logger.error(f'Error publishing log {topic}: {e}') 
```

Reply:
fixed